File: pset 1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string number = get_string("Enter your credit card number: ");
    int len = strlen(number);

    int firsttwo;
    int first;
    int second;
    int sum = 0;
    int multipliedsum = 0;

     for (int i = len - 1, alt = 0; i >= 0; i--, alt++)
    {
        int digit = number[i] - '0';

        if ((alt % 2) == 1)
        {
            multipliedsum = digit * 2;
            sum += (multipliedsum / 10) + (multipliedsum % 10);
        }
        else
        {
            sum += digit;
        }
    }
    first = (number[0] - '0');
    second = (number[1] - '0');
    firsttwo = first * 10 + second;
    if (sum % 10 == 0)
    {
        if ((firsttwo == 34 || firsttwo == 37) && (len == 15))
        {
            printf("AMEX\n");
            return 0;
        }
        else if ((firsttwo >= 51 && firsttwo <= 55) && (len == 16))
        {
            printf("MASTERCARD\n");
            return 0;
        }
        else if ((first == 4) && (len == 13 || len == 16))
        {
            printf("VISA\n");
            return 0;
        }
    }
    printf("INVALID\n");
}

File: pset 6/sentimental-credit/credit.py
from cs50 import get_string


def main():
    number = get_string("Enter your credit card number: ")
    length = len(number)

    firsttwo = 0
    first = 0
    second = 0
    sum = 0
    multipliedsum = 0
    # range (start, stop, step)
    for i in range(length - 1, -1, -1):
        alt = length - 1 - i
        # converts char into int
        digit = int(number[i])

        if ((alt % 2) == 1):

            multipliedsum = digit * 2
            # // used for integer divsion
            sum += (multipliedsum // 10) + (multipliedsum % 10)

        else:
            sum += digit
    first = int((number[0]))
    second = int((number[1]))
    firsttwo = first * 10 + second
    if (sum % 10 == 0):
        if ((firsttwo == 34 or firsttwo == 37) and (length == 15)):
            print("AMEX")
            return

        elif ((firsttwo >= 51 and firsttwo <= 55) and (length == 16)):
            print("MASTERCARD")
            return

        elif ((first == 4) and (length == 13 or length == 16)):
            print("VISA")
            return

    print("INVALID")


main()
